contractcourt/htlc_outgoing_contest_resolver.go

```diff
@@ -155,7 +155,14 @@ func (h *htlcOutgoingContestResolver) Resolve() (ContractResolver, error) {
 			// fail.
 			newHeight := uint32(newBlock.Height)
 			expiry := h.htlcResolution.Expiry
-			if newHeight >= expiry {
+
+			// Check if the expiry height is about to be reached.
+			// We offer this HTLC one block earlier to make sure
+			// when the next block arrives, the sweeper will pick
+			// up this input and sweep it immediately. The sweeper
+			// will handle the waiting for the one last block till
+			// expiry.
+			if newHeight >= expiry-1 {
```

Comment:
what happens if during this last block the preimage confirms, will the sweeper react accordingly and regroup the potential other inputs part of this sweept ?

Reply:
if it's a preimage spent, the timeout resolver will go the preimage path via `claimCleanUp`.

Reply:
so the claimCleanUp cancel this input with the sweeper subsystem in this case, because it was already delivered to the sweeper 1 block earlier ?

Reply:
the sweeper won't attempt the sweeping at `expiry-1`, it just registers it. When one block passes, since the  preimage is received, the channel is not force closed so the input will be attempted and failed.

Reply:
I wonder if we can build in a kind of cancel method, because right now there might be the chance of this input grouping with other inputs, and sweeping will fail although we already know this input is invalid, so we kinda waste one broadcast cycle or ?